mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/configuration/impl/MongoDBConfiguration.java

```diff
@@ -82,14 +86,62 @@ public MongoDBConfiguration(ConfigurationPropertyReader propertyReader, OptionsC
 	 * @return the {@link MongoClientOptions} corresponding to the {@link MongoDBConfiguration}
 	 */
 	public MongoClientOptions buildOptions() {
-		MongoClientOptions.Builder optionsBuilder = new MongoClientOptions.Builder();
+		MongoClientOptions.Builder optionsBuilder = MongoClientOptions.builder();
 
 		optionsBuilder.connectTimeout( timeout );
 		optionsBuilder.writeConcern( writeConcern );
 		optionsBuilder.readPreference( readPreference );
+
+		Map<String, Method> settingsMap = createSettingsMap();
+		for ( Map.Entry<String, Method> entry : settingsMap.entrySet() ) {
+			String setting = MongoDBProperties.MONGO_DRIVER_SETTINGS_PREFIX + "." + entry.getKey();
+			// we know that there is exactly one parameter
+			Class<?> type = entry.getValue().getParameterTypes()[0];
+
+			// for reflection purposes we need to deal with wrapper classes
+			if ( int.class.equals( type ) ) {
+				type = Integer.class;
+			}
+			if ( boolean.class.equals( type ) ) {
+				type = Boolean.class;
+			}
+
+			Object property = propertyReader.property( setting, type ).withDefault( null ).getValue();
+			if ( property == null ) {
+				continue;
+			}
+
+			Method settingMethod = entry.getValue();
+			try {
+				settingMethod.invoke( optionsBuilder, property );
+			}
+			catch ( InvocationTargetException | IllegalAccessException e ) {
+				// TODO OGM-810 fix error message
+			}
+		}
+
 		return optionsBuilder.build();
 	}
 
+	private Map<String, Method> createSettingsMap() {
+		Map<String, Method> settingsMap = new HashMap<>();
+
+		Method[] methods = MongoClientOptions.Builder.class.getDeclaredMethods();
+		for ( Method method : methods ) {
+			if ( method.getParameterTypes().length == 1 ) {
+				Class<?> parameterType = method.getParameterTypes()[0];
+				// we just care of string, int and boolean setters
+				if ( String.class.equals( parameterType )
+						|| int.class.equals( parameterType )
```

Comment:
@hferentschik You think you could do that as part of this PR? If not, I'll open another follow-up just for that one, as it can be done separately from (and much quicker than) addressing the deprecation issues.